### PR TITLE
feat: implement alternative React hooks

### PR DESCRIPTION
**Pull request by morgante**
Implements some basics to allow branching hook logic. Will follow up with another PR.

### CODE REVIEW

File: .grit/patterns/react_to_hooks.md [hunk 1]
```diff
@@ -10,7 +10,45 @@ tags: #react, #migration, #complex
...
+                  predefined_type()
+              } as $inner_type
+          }
+        },
+        // We have our type and our ref, so now create the statement
+        $statements += `const $name = useRef<$inner_type>($value);`
+      },
+      // If type is not defined, we can't use a type annotation
+      and {
+        // Type is still not defined, try to find it from the statement
```
Comment (seren5240) [inline .grit/patterns/react_to_hooks.md:41]:
```suggestion
        // Type can't be located, initialize `useRef` without a type
```

File: .grit/patterns/react_to_hooks.md [hunk 2]
```diff
@@ -50,7 +88,11 @@ pattern handle_one_statement($class_name, $statements, $states_statements, $stat
             and {
                 $statement <: prepend_comment($statements),
                 $async <: `async`,
-                $statements += `const ${name}Handler = useCallback(async $parameters => $body, []);`
+                if ($handler_callback_suffix <: .) {
+                  $statements += `const ${name} = async $parameters => $body;`
+                } else {
+                  $statements += `const ${name}${handler_callback_suffix} = useCallback(async $parameters => $body, []);`
```
Comment (seren5240) [inline .grit/patterns/react_to_hooks.md:94]: From my understanding one of the alternative cases we'd have to support is that these shouldn't become `useCallback`s at all but just bare functions, or are we not doing that?

File: .grit/patterns/react_to_hooks.md [hunk 3]
```diff
@@ -327,9 +377,12 @@ pattern first_step() {
...
         $the_function = `($args) => {\n$constructor_statements\n\n    $states_statements\n\n    ${statements}\n\n    ${render_statements} \n}`,
 
-
-        if ($body <: contains `ViewState`) {
-            $the_const = `import { observer } from "mobx-react";\n\nconst $class_name$const_type_annotation = observer($the_function);`
+        // Construct the final class name
+        $original_name = $class_name,
+        if ($body <: contains r"(v|V)iewState"($_)) {
+            $base_name = js"${class_name}Base",
+            $the_const = `const $base_name$const_type_annotation = $the_function;
```
Comment (seren5240) [inline .grit/patterns/react_to_hooks.md:384]:
```suggestion
            $the_const = `const $base_name$const_type_annotation = $the_function;
            
```

Leave an empty line for formatting?

**Review by seren5240 — APPROVED** [review]

File: .grit/patterns/react_to_hooks.md (hunk 2 above)
Comment (morgante) [reply .grit/patterns/react_to_hooks.md:93]: That's what setting `handler_callback_suffix` to null is fore.

**Pull request merged**